phd-tests/framework/src/guest_os/windows.rs

```diff
@@ -0,0 +1,95 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Helper functions for generating Windows guest OS adaptations.
+
+use super::{CommandSequence, CommandSequenceEntry, GuestOsKind};
+
+/// Emits the login seqeunce for the given `guest`, which must be one of the
+/// Windows guest OS flavors.
+///
+/// This login sequence assumes the following:
+///
+/// - The image has been generalized (by running `sysprep /generalize`) and is
+///   configured so that on first boot it will skip the out-of-box experience
+///   (OOBE) and initialize the local administrator account with the appropriate
+///   password. See [MSDN's Windows Setup
+///   documentation](https://learn.microsoft.com/en-us/windows-hardware/manufacture/desktop/generalize?view=windows-11)
+///   for more details.
+/// - Cygwin is installed to C:\cygwin and can be launched by invoking
+///   C:\cygwin\cygwin.bat.
+/// - The local administrator account is enabled with password `0xide#1Fan`.
+pub(super) fn get_login_sequence_for(guest: GuestOsKind) -> CommandSequence {
+    assert!(matches!(
+        guest,
+        GuestOsKind::WindowsServer2019 | GuestOsKind::WindowsServer2022
+    ));
```

Comment:
nitpicky and very much not important: we _could_ avoid having to do this if we had a 
```rust
enum GuestOsKind {
    Windows(Windows), 
    // ...
}

enum Windows {
    Server2019,
    Server2022,
}
```
but i'm not sure if it's worth bothering that or not, up to you.

Reply:
yeah, that was what i was imagining my suggestion would grow into, eventually. i’m fine with saving it for a later PR!